Character controller

Codidodido opened the pull request.

Codidodido committed:
Create player movement script

Run & Jump

Codidodido committed:
Make mario's idle, run & jump animation

Codidodido committed:
Modify Player movement

Handle animations, flip player & refactor

Codidodido committed:
Create mario dead animation

Codidodido committed:
Modify player movement

Lose game when colliding to enemy

Codidodido committed:
Modify player movement and create lose script

Transform lose dynamic from player movement to lose script and attach it to Mario

Codidodido committed:
Add mono behaviour singleton script

Codidodido committed:
Create time script

Start the game with a countdown

Codidodido committed:
Modify player movement script

Enable to move when timer event 'on finish count down' is invoke

Codidodido committed:
Create UI countdown timer

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,113 @@
+
+    private void HandleAnimation()
+    {
+        if (_xVector != 0 && !_animator.GetBool("IsJump"))
+        {
+            _animator.SetFloat("Speed", 1);
+        }
+        else
+        {
+            _animator.SetFloat("Speed", 0);
+        }
```
```suggestion
        if (_xVector != 0 && !_animator.GetBool("IsJump"))
            _animator.SetFloat("Speed", 1);
        else
            _animator.SetFloat("Speed", 0);
```

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,113 @@
+            _animator.SetFloat("Speed", 0);
+        }
+
+        if (!CheckDistanceFromEarth())
+        {
+            _animator.SetBool("IsJump", true);
+        }
+        else
+        {
+            _animator.SetBool("IsJump", false);
+        }
```
```suggestion
        if (!CheckDistanceFromEarth())
            _animator.SetBool("IsJump", true);
        else
            _animator.SetBool("IsJump", false);
```

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,113 @@
+
+    }
+
+    private bool CheckDistanceFromEarth()
```
rename the function to something that should return a boolean, something like `IsOnGround()`

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,113 @@
+    {
+        RaycastHit2D hit;
+        hit = Physics2D.Raycast(transform.position, -transform.up, _distanceFromEarth, LayerMask.GetMask("Ground"));
+        if (hit.collider != null)
+        {
+            return true;
+        }
+        else
+        {
+            return false;
+        }
```
```suggestion
        if (hit.collider != null)
            return true;
        else
            return false;
```

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,113 @@
+
+    private void OnEnable()
+    {
+        Timer.Instance.onFinishCountDown += EnableMoving;
```
singletons should not be that common in your code

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,113 @@
+
+    private void Jump()
+    {
+        if (Input.GetKeyDown(KeyCode.Space) && CheckDistanceFromEarth() && _moveable)
```
when you say jump, it should just fucking jump

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,113 @@
+
+    private void Run()
+    {
+        if (_animator.GetBool("IsAlive") && _moveable)
```
and this

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,113 @@
+        {
+            _xVector = Input.GetAxisRaw("Horizontal");
+            transform.position += (Vector3)new Vector2(_xVector * moveSpeed * Time.deltaTime, 0);
+            Flip();
```
Change this into a `ChangeScale(Vector2 value)` and remove the if inside the function

Codidodido committed:
Refactor player movement script

Edit methods, change names and remove useless codes

Codidodido committed:
Edit count down dynamic

Remove singleton behaviour from timer and handle timer dynamic directly through player movement script

Codidodido committed:
Merge branch 'Timer' into Character-Controller

Codidodido committed:
Refactor timer script

URAmiRBin commented:
resolve the conflicts and mention me

Codidodido committed:
Merge branch 'master' into Character-Controller

# Conflicts:
-	Assets/Scenes/SampleScene.unity

Codidodido committed:
Import cinemachine package
